src/test/groovy/org/prebid/server/functional/tests/privacy/CcpaAuctionSpec.groovy

```diff
@@ -122,76 +124,101 @@ class CcpaSpec extends PrivacyBaseSpec {
+        def bidderRequests = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequests.device?.geo == maskGeo(bidRequest)
 
-        verifyAll {
-            privacy.originPrivacy?.ccpa?.usPrivacy == validCcpa as String
-            privacy.resolvedPrivacy?.ccpa?.usPrivacy == validCcpa as String
+        where:
+        ccpaConfig << [new AccountCcpaConfig(enabled: false, channelEnabled: [(ChannelType.APP): true]),
+                       new AccountCcpaConfig(enabled: true)]
+    }
 
-            !privacy.originPrivacy?.coppa?.coppa
-            !privacy.resolvedPrivacy?.coppa?.coppa
+    @Unroll
+    def "PBS should apply ccpa when privacy.ccpa.channel-enabled.web or privacy.ccpa.enabled = true in account config"() {
+        given: "Default basic generic BidRequest"
+        def validCcpa = new CcpaConsent(explicitNotice: ENFORCED, optOutSale: ENFORCED)
+        def bidRequest = getCcpaBidRequest(validCcpa)
 
-            !privacy.originPrivacy?.tcf?.gdpr
-            !privacy.originPrivacy?.tcf?.tcfConsentString
-            !privacy.originPrivacy?.tcf?.tcfConsentVersion
-            !privacy.originPrivacy?.tcf?.inEea
-            !privacy.resolvedPrivacy?.tcf?.gdpr
-            !privacy.resolvedPrivacy?.tcf?.tcfConsentString
-            !privacy.resolvedPrivacy?.tcf?.tcfConsentVersion
-            !privacy.resolvedPrivacy?.tcf?.inEea
+        and: "Save account config into DB"
+        def privacy = new AccountPrivacyConfig(ccpa: ccpaConfig)
+        def accountConfig = new AccountConfig(privacy: privacy)
+        def account = new Account(uuid: bidRequest.site.publisher.id, config: accountConfig)
+        accountDao.save(account)
 
-            privacy.privacyActionsPerBidder[BidderName.GENERIC] ==
-                    ["Geolocation was masked in request to bidder according to CCPA policy."]
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
 
-            privacy.errors?.isEmpty()
-        }
+        then: "Bidder request should contain masked values"
+        def bidderRequests = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequests.device?.geo == maskGeo(bidRequest)
+
+        where:
+        ccpaConfig << [new AccountCcpaConfig(enabled: false, channelEnabled: [(ChannelType.WEB): true]),
+                new AccountCcpaConfig(enabled: true)]
     }
 
-    @PendingFeature
-    def "PBS should add debug log for amp request when invalid ccpa was passed"() {
-        given: "Default AmpRequest"
-        def invalidCcpa = new BogusConsent()
-        def ampRequest = getCcpaAmpRequest(invalidCcpa)
+    @Unroll
+    def "PBS should not apply ccpa when privacy.ccpa.channel-enabled.app or privacy.ccpa.enabled = false in account config"() {
+        given: "Default basic generic BidRequest"
+        def validCcpa = new CcpaConsent(explicitNotice: ENFORCED, optOutSale: ENFORCED)
+        def bidRequest = getCcpaBidRequest(DistributionChannel.APP, validCcpa)
```

Comment:
you may static import enum (take a look pls through the same occurrences in that PR)

Reply:
added static import everywhere except APP, because there are several enams in the tests with the same value (DistributionChannel and ChannelType)